### PR TITLE
Introduce Merge Conflict Check

### PR DESCRIPTION
**Pull request by lkiesow**
This patch transfers the very convenient merge conflict check from
Opencast to Tobira.

### CODE REVIEW

File: .github/workflows/check-merge-conflict.yml
Comment (JulianKniephoff) [inline .github/workflows/check-merge-conflict.yml:5]: The docs are a bit sparse on this: Does `synchronize` also trigger when a PR is first created? I would think so, right? :thinking: But if not, we probably want another trigger for that event.

File: .github/workflows/check-merge-conflict.yml
Comment (JulianKniephoff) [inline .github/workflows/check-merge-conflict.yml:14]: Should this fit in our existing label schema? I'm thinking sth. like `status:conflicts`. @LukasKalbertodt, you are the label master. Opinions?

File: .github/workflows/check-merge-conflict.yml
Comment (JulianKniephoff) [inline .github/workflows/check-merge-conflict.yml:16]: Should we also `commentOnClean`? Or do we notice that by the removal of the label, alone? Are there notifications for label events? :thinking: Do we even want that?! :scream: 

No hard opinions here, just thinking about how we will use this.

File: .github/workflows/check-merge-conflict.yml
Comment (JulianKniephoff) [inline .github/workflows/check-merge-conflict.yml:11]:
```suggestion
      - name: Check for dirty PRs
```

I would prefer this phrasing, but that's not the point here. The fact that the workflow name has proper casing, but this doesn't triggers me. :see_no_evil:

**Review by JulianKniephoff — COMMENTED** [review]
A minor nitpick and some discussion points. Nothing that would block this PR, though.

File: .github/workflows/check-merge-conflict.yml
Comment (LukasKalbertodt) [reply .github/workflows/check-merge-conflict.yml:14]: Label master :D 

But yes, `status:conflicts` sounds good. I just created that label.

File: .github/workflows/check-merge-conflict.yml
Comment (LukasKalbertodt) [reply .github/workflows/check-merge-conflict.yml:16]: Almost always, the conflicts will be resolved by the author pushing to the branch. So the author is already active at that time (unlike when the conflicts arise). So I think it's fine not to comment and not to notify.

Comment (JulianKniephoff) [reply .github/workflows/check-merge-conflict.yml:16]: The notifications would be for potential reviewers in that case, though.

Comment (LukasKalbertodt) [inline .github/workflows/check-merge-conflict.yml:16]: Sounds a bit formal, no? :P Just a stupid suggestion.

```suggestion
          commentOnDirty: "This pull request has conflicts :pensive: Please resolve those before we can review the PR. Thanks!"
```

**Review by LukasKalbertodt — COMMENTED** [review]
I don't know that much about GH actions yet and since Julian apparently did a technical review already, let me just say: I am all in favor of adding this. Seems very useful ^_^

File: .github/workflows/check-merge-conflict.yml
Comment (LukasKalbertodt) [reply .github/workflows/check-merge-conflict.yml:16]: Reviewers get notified when someone pushes on the branch, right?

File: .github/workflows/check-merge-conflict.yml
Comment (lkiesow) [reply .github/workflows/check-merge-conflict.yml:11]: The, let's go for
```suggestion
      - name: check for dirty pull requests
```

File: .github/workflows/check-merge-conflict.yml
Comment (JulianKniephoff) [reply .github/workflows/check-merge-conflict.yml:16]: For ppl who are lazy like me: This is what `:pensive:` looks like: :pensive:

Comment (lkiesow) [reply .github/workflows/check-merge-conflict.yml:16]: Yes, they would.
I also don't think adding more notifications is really helpful when we know that they are almost universally not helpful.
We would just start ignoring them

File: .github/workflows/check-merge-conflict.yml
Comment (lkiesow) [reply .github/workflows/check-merge-conflict.yml:14]: fine with me. Will use that

File: .github/workflows/check-merge-conflict.yml
Comment (JulianKniephoff) [reply .github/workflows/check-merge-conflict.yml:11]: But that's still cased inconsistently with the workflow `name`! Where is my `:triggered:` emoji?!

File: .github/workflows/check-merge-conflict.yml
Comment (JulianKniephoff) [reply .github/workflows/check-merge-conflict.yml:16]: Fair enough.

Comment (lkiesow) [reply .github/workflows/check-merge-conflict.yml:16]: Let's keep this informative and neutral (no exclamations) and let's not use inconsistently cased abbreviations.

```suggestion
          commentOnDirty: |
              This pull request has conflicts.
              Please resolve those so we can review the pull request.
              Thanks.
```
I don't like the smiley in here. I think it's actually somewhat ridiculous. But if you must have it

```suggestion
          commentOnDirty: |
              This pull request has conflicts :pensive:
              Please resolve those so we can review the pull request.
              Thanks.
```

Comment (LukasKalbertodt) [reply .github/workflows/check-merge-conflict.yml:16]: I don't have a strong opinion here :D 
Your call.

File: .github/workflows/check-merge-conflict.yml
Comment (lkiesow) [reply .github/workflows/check-merge-conflict.yml:5]: I don't think it is but GitHub will tell you that your pull request has conflict while you are creating it. Do we want a bot that immediately tells you the same thing?

If so, we probably want to set:

```yml
  pull_request_target:
    types:
      - opened
      - synchronize
```


The events [should be triggered](https://frontside.com/blog/2020-05-26-github-actions-pull_request/#how-do-workflows-trigger-on-pull_request) when…
- opened: pull request is created
- synchronize: commit(s) pushed to the pull request

File: .github/workflows/check-merge-conflict.yml
Comment (lkiesow) [reply .github/workflows/check-merge-conflict.yml:16]: @JulianKniephoff, you have the remainder of the day to click on one suggestion – or to complain and suggest a new one – otherwise I'll pick one of those.

File: .github/workflows/check-merge-conflict.yml
Comment (lkiesow) [reply .github/workflows/check-merge-conflict.yml]: I hope it's fine like this.

File: .github/workflows/check-merge-conflict.yml
Comment (JulianKniephoff) [reply .github/workflows/check-merge-conflict.yml:5]: You have a good point there, maybe we don't need it. You decide, I'm fine either way. :)

File: .github/workflows/check-merge-conflict.yml
Comment (JulianKniephoff) [reply .github/workflows/check-merge-conflict.yml:16]: I actually can't click a button to accept the suggestions, so I'm just giving my vague opinion: I like emoji in online communication a lot, so I would tend to include one as well. I don't think :pensive: is the right one, though. I would probably just go for :frowning_face:. I can live with not having either of them as well, though.

Comment (LukasKalbertodt) [reply .github/workflows/check-merge-conflict.yml:16]: I also do like emoji in online communication a lot. But I also get what you mean by "ridiculous". For example, I saw [this](https://medium.com/styled-components/announcing-styled-components-v5-beast-mode-389747abd987) release announcement recently and cringed a bit :P 

Just to add another data point: Rust bots (including the one that notifies of merge conflicts) use emojis, but not faces. For example :cloud_with_rain: or :umbrella: for merge conflicts, :evergreen_tree: for another thing. Dunno if that's any better.

File: .github/workflows/check-merge-conflict.yml
Comment (lkiesow) [reply .github/workflows/check-merge-conflict.yml]: It's now `opened` and `synchronize`

File: .github/workflows/check-merge-conflict.yml
Comment (lkiesow) [reply .github/workflows/check-merge-conflict.yml]: Updated this and set a frowning face emoji

**Review by JulianKniephoff — APPROVED** [review]

**Comment by LukasKalbertodt** [comment]
Wohoo, it works: https://github.com/elan-ev/tobira/pull/53#issuecomment-715343767
:tada:

**Pull request merged**
